### ISSUE TITLE
Crash crystal tool format with heredocs, interpolation, and a square brace 

### ISSUE THREAD

**Issue by fdr** [crystal-lang/crystal#9888]
Source text looks like this:
```crystal
var = "hello world"
puts <<-EOS
[
#{var}
EOS
```
It does what you expect:

```
crystal run bug.cr
[
hello world
```

But crystal tool format crashes like this:

```
crystal tool format bug.cr --show-backtrace
expecting DELIMITER_START, not `IDENT, var`, at :4:3 (Exception)
  from ../../../../../crystal/src/io/memory.cr:427:13 in 'visit'
  from ../../../../../crystal/src/compiler/crystal/tools/formatter.cr:160:9 in 'visit_string_interpolation'
  from ../../../../../crystal/src/compiler/crystal/tools/formatter.cr:648:13 in 'next_token'
  from ../../../../../crystal/src/slice.cr:229:5 in 'accept'
  from ../../../../../crystal/src/compiler/crystal/syntax/visitor.cr:20:7 in 'format_args_simple'
  from ../../../../../crystal/src/compiler/crystal/tools/formatter.cr:2719:54 in 'format_args'
  from ../../../../../crystal/src/compiler/crystal/tools/formatter.cr:2713:5 in 'visit'
  from ../../../../../crystal/src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from ../../../../../crystal/src/compiler/crystal/syntax/visitor.cr:20:7 in 'accept'
  from ../../../../../crystal/src/compiler/crystal/tools/formatter.cr:17:7 in 'format_source'
  from ../../../../../crystal/src/compiler/crystal/command/format.cr:103:9 in 'format_many'
  from ../../../../../crystal/src/compiler/crystal/command/format.cr:53:5 in 'format'
  from ../../../../../crystal/src/hash.cr:896:11 in 'tool'
  from ../../../../../crystal/src/compiler/crystal/command.cr:107:7 in 'run'
  from ../../../../../crystal/src/compiler/crystal.cr:11:1 in '__crystal_main'
  from ../../../../../crystal/src/crystal/main.cr:105:5 in 'main'
  from src/env/__libc_start_main.c:94:2 in 'libc_start_main_stage2'
```

**Comment by MakeNowJust** [comment]
It is the same problem as #9473 and fixed in #9475.
In fact, this does not cause on the HEAD compiler. So, we can close this.

**Comment by asterite** [comment]
Thank you @MakeNowJust
